Orange/widgets/utils/scaling.py

```diff
@@ -30,53 +31,59 @@ def rescale_data(self):
 
     def _compute_domain_data_stat(self):
         stt = self.domain_data_stat = \
-            getCached(self.data, DomainBasicStats, (self.data,))
-        for index in range(len(self.domain)):
-            attr = self.domain[index]
+            getCached(self.data, DomainBasicStats, (self.data, True))
+        domain = self.domain
+        for attr in domain.attributes + domain.class_vars + domain.metas:
```

Comment:
Maybe `chain(domain, domain.metas)`?

Reply:
To be sincere, I would prefer `domain.all_variables` or something similar. Like we have `domain.variables = domain.attributes + domain.class_vars` we could have something that includes `attributes`, `class_vars` and `metas`.

Reply:
I hate this, too. 

When one gets older, (s)he start adding historic notes to everything. So let me say that in the beginning, like 20 years ago, Orange used to have attributes and classes, and both together were called variables. Then I introduced meta attributes, and they didn't mean the same thing as today: meta attributes were (essentially) sparse, while attributes and classes were dense ("essentially", because there was no numpy, these were C++ structures). Since all code - before the introduction of metas - expected that variables were attributes + classes, I would break everything if I added metas there. It also made more sense than today since variables (=attributes + classes) were the dense part of the data.

The idea that any part can be dense or sparse, and that metas are not so different from attributes and so forth evolved gradually even in Orange 3. Hence we have not initially decided that `variables` will be attributes + class + meta. To make things worse, we (most probably: I) extended this bad design to iteration over `Domain`. @kernc (I think) correctly pointed out that since `var in domain` returns `True` also when `var` is in `domain.metas`, so `for var in domain` should iterate over attributes, classes and also metas. So in all the places where you could use `chain(domain, domain.metas)` in your code, just `domain` should suffice instead. Alas, fixing the iteration over `Domain` would break so much code that we gave up. This horrible design decision stays forever, I fear.

I don't like `all_variables`. `variables` should already be *all*, so `all_variables` sounds almost like `really_all_variables`. Plus, the next thing you'll want will be `primitive_variables`. And maybe also `continuous_variables`. I wouldn't like to multiply the number of domains properties.

Reply:
For the sake of brainstorming - and if this is indeed necessary - what about having a method `Domain.all(condition)`. If `condition` is a type derived from `Variable`, `all` returns all variables of that type (and derived types). Hence , `Domain.all(DiscreteVariable)` returns all discrete variables. We then add `PrimitiveVariable` as a new common base class for `DiscreteVariable` and `ContinuousVariable`. With that, `Domain.all(PrimitiveVariable)` returns all primitive variables. Finally, the default value for `condition` is `Variable`, so `Domain.all()` returns attributes + class_vars + metas.

Method `all` could also accept other types of conditions if we can think of any.

Just for brainstorming.

Reply:
Great idea, I implemented it.